addons/binding/org.openhab.binding.nikohomecontrol/src/main/java/org/openhab/binding/nikohomecontrol/handler/NikoHomeControlBridgeHandler.java

```diff
@@ -0,0 +1,238 @@
+        // There is nothing to handle in the bridge handler
+    }
+
+    @Override
+    public void initialize() {
+
+        logger.debug("Niko Home Control: initializing bridge handler");
+
+        Configuration config = this.getConfig();
+
+        port = ((Number) config.get(CONFIG_PORT)).intValue();
+        logger.debug("Niko Home Control: bridge handler port {}", port);
+
+        try {
+            // If hostname or address was provided in the configuration, try to use this to for bridge and give error
+            // when hostname parameter was not valid.
+            // No hostname provided is a valid configuration, therefore allow null addr to pass through.
+            if (config.get(CONFIG_HOST_NAME) != null) {
+                logger.debug("Niko Home Control: bridge handler host {}", config.get(CONFIG_HOST_NAME));
+                addr = InetAddress.getByName((String) config.get(CONFIG_HOST_NAME));
+            }
+        } catch (UnknownHostException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
+                    "Niko Home Control: cannot resolve bridge IP with hostname " + config.get(CONFIG_HOST_NAME));
+            return;
+        }
+
+        Integer refreshInterval = ((Number) config.get(CONFIG_REFRESH)).intValue();
+
+        createCommunicationObject();
+        setupRefreshTimer(refreshInterval);
+
+    }
+
+    /**
+     * Create communication object to Niko Home Control gateway and start communication.
+     * Trigger discovery when communication setup is successful.
+     *
+     * @param nhcAddr IP address or subnet for gateway
+     * @param port
+     */
+    private void createCommunicationObject() {
+
+        scheduler.submit(new Runnable() {
+
+            @Override
+            public void run() {
+                try {
+                    for (int i = 0; i < 3; i++) {
+                        // try connecting max 3 times
+                        nhcComm = new NikoHomeControlCommunication(addr, port);
+                        if (nhcComm.communicationActive()) {
+                            break;
+                        }
+                        if (i == 2) {
+                            throw new IOException("Niko Home Control: communication socket error");
+                        }
+                    }
+
+                    addr = nhcComm.getAddr();
+                    // keep the discovered host ip in the configuration to reuse when restarting
+                    thing.getConfiguration().put(CONFIG_HOST_NAME, addr.getHostAddress());
+
+                    updateProperties();
+
+                    updateStatus(ThingStatus.ONLINE);
+
+                    if (nhcDiscovery != null) {
+                        nhcDiscovery.discoverDevices();
+                    } else {
+                        logger.debug("Niko Home Control: cannot discover, discovery service not started");
+                    }
+
+                } catch (IOException e) {
```

Comment:
I need to correct myself here. It is thrown from current line 123 (was 111) and line 119. Both of them are communication issues, so that's why I catch both of them together. The debug logs would show the underlying difference.